actor/todo-sql/README.md

```diff
@@ -9,26 +9,82 @@ and the logging capability (`wasmcloud:logging`).
+                                            |
+                                            |- hostspec
+```
 
-- If you want to change from the default https port 9000, edit http-linkdef.json to change the port in the address line and also in the CORS origins
-- set up postgres database with a user that has write access (can create tables, insert, and select)
-- edit sql-linkdef.json and update the uri field with the credentials and host/port
-- make sure nats and the wasmcloud host are running
-- run `make clean-start` to push the actor, start the providers, and link neverything
-- Open your browser to https://localhost:9000/", and accept the warning from the browser that the server is using a self-signed certificate.
+## Step 3 -  Running WasmCloud and NATS
+- Follow the guide for installing [WasmCloud](https://wasmcloud.dev/overview/installation/) and [NATS](https://docs.nats.io/running-a-nats-service/introduction/installation).
+- Also make sure that you have [jq](https://stedolan.github.io/jq/) as well as Rust installed.
+- Run the WasmCloud host by running `wash up --allowed-insecure localhost:5000,127.0.0.1:5000` since we are using a self-signed certificate.
+- Navigate to `localhost:4000` to view the WasmCloud dashboard.
 
-## to run the todo test suite (from www.todobackend.com)
+## Step 4 - Run the Makefile and Tests
+- Run `make clean-start` from the todo-sql folder. This will push the actor, start the providers, create the necessary linkdefs and link the components with each other. This is how your WasmCloud dashboard should look like:
+![WasmCloud Dashboard](images/wasmcloud-dashboard.png)
+- Open your browser to "https://localhost:9000/", and accept the warning from the browser that the server is using a self-signed certificate.
+- Let's run the test suite from www.todobackend.com by navigating to "https://localhost:9000/test/" and clicking on the green button. You should see the tests cleared: 
+![Tests cleared on localhost:9000](images/tests-passed.png)
+- Congratulations you have run the postgres example successfully.
 
-- visit "https://localhost:9000/test/" to view the todo test page and press the green button"
\ No newline at end of file
+## Troubleshooting
+- Make sure all the ports that are being used are open : 4000, 5000, 5432 and 9000.
+- Docker should have permission to access the examples folder and its subdirectories as it needs access to the `create-tables.sql file`.
+- Check the paths in `http-linkdef.json` and uri in `sql-linkdef.json`. Check if your postgres database is working in the specific docker container and that the correct user has all the necessary permissions.
+- `wash up` uses port 4000 as of now. If the host is unable to start, inspect this port and kill the necessary processes before restarting. `wash down` will gracefully shut down the host as well.
+- If you see a `base64: invalid argument -w0` in your logs, change `base64 -w0` to `base64 -i` in your Makefile.
```

Comment:
Ideally the makefile should work on different platforms. I _think_ this change to the makefile would work with both mac and GNU versions of base64:
```bash
base64 -w0 sql-linkdef.json 2>/dev/null || base64 -i sql-linkdef.json 2>/dev/null
```

It's a bit hacky, but better than assuming the user has a specific version installed

Reply:
Yes this works. Changed it in the makefile.